src/analysis/processing/qgsalgorithmfillnodata.cpp

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+                         qgsalgorithmfillnodata.cpp
+                         ---------------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Clemens Raffler
+    email                : clemens dot raffler at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmfillnodata.h"
+#include "qgsrasterfilewriter.h"
+
+///@cond PRIVATE
+
+QString QgsFillNoDataAlgorithm::name() const
+{
+  return QStringLiteral( "fillnodata" );
+}
+
+QString QgsFillNoDataAlgorithm::displayName() const
+{
+  return QObject::tr( "Fill NoData cells" );
+}
+
+QStringList QgsFillNoDataAlgorithm::tags() const
+{
+  return QObject::tr( "noData,NoData,data,cells,fill,set" ).split( ',' );
```

Comment:
```suggestion
  return QObject::tr( "noData,NoData,data,cells,fill,set" ).split( ',' );
```
```suggestion
  return QObject::tr( "data,cells,fill,set" ).split( ',' );
```

The tag searching isn't case sensitive, and there's no need to include terms which are already part of the display name.